### ISSUE TITLE
Can configBuilders be used with system.web config sections?

### ISSUE THREAD

**Issue by nickwesselman** [aspnet/MicrosoftConfigurationBuilders#149]
I have not been able to find a clear answer to this question. I'm trying to implement a `SectionHandler` for `CustomErrorsSection`, but it seems like the `configBuilders` attribute on `customErrors` is being ignored entirely.

```csharp
    public class CustomErrorsSectionHandler : Microsoft.Configuration.ConfigurationBuilders.SectionHandler<CustomErrorsSection>
    {
        private static readonly string _modeKey = "MODE";

        public override IEnumerator<KeyValuePair<string, object>> GetEnumerator()
        {
            yield return new KeyValuePair<string, object>(_modeKey, ConfigSection.Mode);
        }

        public override void InsertOrUpdate(string newKey, string newValue, string oldKey = null, object oldItem = null)
        {
            if (string.IsNullOrEmpty(newValue))
            {
                return;
            }

            if (_modeKey.Equals(newKey, StringComparison.InvariantCultureIgnoreCase))
            {
                ConfigSection.Mode = (CustomErrorsMode)Enum.Parse(typeof(CustomErrorsMode), newValue);
            }
        }
    }
```

With my configuration isolated in a transform:
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
  <configSections>
    <section name="Microsoft.Configuration.ConfigurationBuilders.SectionHandlers"
             type="Microsoft.Configuration.ConfigurationBuilders.SectionHandlersSection, Microsoft.Configuration.ConfigurationBuilders.Base"
             restartOnExternalChanges="false"
             requirePermission="false"
             xdt:Transform="InsertIfMissing"
             xdt:Locator="Match(name)" />
  </configSections>
  <Microsoft.Configuration.ConfigurationBuilders.SectionHandlers xdt:Transform="InsertIfMissing">
    <handlers>
      <add name="CustomErrorsSectionHandler"
           type="SectionHandlerTest.CustomErrorsSectionHandler, SectionHandlerTest"
           xdt:Transform="InsertIfMissing"
           xdt:Locator="Match(name)" />
    </handlers>
  </Microsoft.Configuration.ConfigurationBuilders.SectionHandlers>
  <configBuilders xdt:Transform="InsertIfMissing">
    <builders xdt:Transform="InsertIfMissing">
      <add name="CustomErrorsBuilder"
           mode="Strict"
           prefix="CUSTOM_ERRORS_"
           stripPrefix="true"
           type="Microsoft.Configuration.ConfigurationBuilders.EnvironmentConfigBuilder, Microsoft.Configuration.ConfigurationBuilders.Environment"
           xdt:Transform="InsertIfMissing"
           xdt:Locator="Match(name)" />
    </builders>
  </configBuilders>
  <system.web xdt:Transform="InsertIfMissing">
    <customErrors configBuilders="CustomErrorsBuilder"
                  xdt:Transform="SetAttributes" />
  </system.web>
</configuration>
```

**Comment by nickwesselman** [comment]
Got this working, apparently my XDT wasn't applying and/or I was not adding the needed config elements in the right order.

**Comment by nickwesselman** [comment]
Added note though -- this cannot be used with the `<compilation />` section, it appears that it's loaded before the app domain (which makes sense I suppose), so unless config builders and other required assemblies are in the GAC, it will not work.

**Comment by StephenMolloy** [comment]
Glad you got it working @nickwesselman.

Regarding use on the `compilation` section... you might be able to get it to work somewhat. For non-ASP.Net-based apps, loading config is relatively straight forward. It just happens, and the config builders run as part of the config loading process. But in ASP.Net, because it is a multi-tenant host, things work a little bit differently.

When a request comes in to the web server, it initially goes through the default app domain where ASP.Net determines whether it needs to spin up a new AppDomain for the application, or if one already exists that it can pass the request through. When spinning up new app domains, ASP.Net first reads a few settings _from the applications web.config_ but reads them _in the default app domain._

Reading this handful of settings still goes through the config system... but the bin path of the default domain does not include the application's bin folder. So if your config builder is deployed in your application, it won't be found. Normally this would result in a type load exception. But in this special cross-domain case, we have added some logic to silently ignore config builders that can't be found.

Long story short - if you put your config builder assemblies in the GAC, then you might be able to use them on `<compilation />`.

**Comment by nickwesselman** [comment]
@StephenMolloy Thanks for the details, makes sense. In our case, getting things into the GAC is a big lift, so we went down [another path](Sitecore/docker-tools#4) for now, but good to know it's possible.
